testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -344,31 +352,108 @@ async fn wait_genesis_job(kube_client: &K8sClient, era: usize) -> Result<(), any
     .await
 }
 
-pub fn clean_k8s_cluster(
-    helm_repo: String,
-    base_num_validators: usize,
-) -> Result<(), anyhow::Error> {
-    assert!(base_num_validators <= MAX_NUM_VALIDATORS);
+fn remove_validator(validator_name: &str) -> Result<()> {
+    let validator_uninstall_args = ["uninstall", "--keep-history", validator_name];
+    println!("{:?}", validator_uninstall_args);
+    let validator_uninstall_output = Command::new(HELM_BIN)
+        .stdout(Stdio::inherit())
+        .args(&validator_uninstall_args)
+        .output()
+        .expect("failed to helm uninstall valNN");
 
-    // get the previous chain era
+    let uninstalled_re = Regex::new(r"already deleted").unwrap();
+    let uninstall_stderr = String::from_utf8(validator_uninstall_output.stderr).unwrap();
+    let already_uninstalled = uninstalled_re.is_match(&uninstall_stderr);
+    assert!(
+        validator_uninstall_output.status.success() || already_uninstalled,
+        "{}",
+        uninstall_stderr
+    );
```

Comment:
maybe a log here to indicate remove work is done. same as below for validator upgrading